sopel/tools/identifiers.py

```diff
@@ -0,0 +1,268 @@
+"""Identifier tools to represent IRC names (nick or channel).
+
+Nick and channel are defined by their names, which are "identifiers": their
+names are used to differentiate users from each others, channels from each
+others. To ensure that two channels or two users are the same, their
+identifiers must be processed to be compared properly. This process depends on
+which RFC and how that RFC is implemented by the server: IRC being an old
+protocol, different RFCs have differents version of that process:
+
+* :rfc:`1549#section-2.2`: ASCII characters, and ``[]\\`` are mapped to ``{}|``
+* :rfc:`2812#section-2.2`: same as in the previous RFC, adding ``~`` mapped to
+  ``^``
```

Comment:
```suggestion
* :rfc:`RFC 1459 § 2.2<1459#section-2.2>`: ASCII characters, and ``[]\\`` are
  mapped to ``{}|``
* :rfc:`RFC 2812 § 2.2<2812#section-2.2>`: same as in the previous RFC, adding
  ``~`` mapped to ``^``
```

Bonus catch: transposed digits. We're not talking about [PPP in HDLC Framing](https://datatracker.ietf.org/doc/html/rfc1549.html). 😄